test/cases/schema-title-as-type-name.ts

```diff
@@ -0,0 +1,48 @@
+export var schema = {
+    "$schema": "http://json-schema.org/draft-03/schema",
+    "id": "http://mycompany.com/api/referencing.json",
+    "title": "Referencing",
+    "type": "object",
+    "properties": {
+        "ref": {
+            "$ref": "../../test/resources/ReferencedType.json"
```

Comment:
That is what I did in the beginning, but find out that there is two different code paths depending if you are compiling a top level definition or a referenced one.

See https://github.com/bcherny/json-schema-to-typescript/pull/54/commits/833a0a11532137d79ecb5d171976fc343404bf9f#diff-f41e9d04a45c83f3b6f6e630f10117feL259

That's why I explicitly added the reference in the test case.